lang/java/tools/src/main/java/org/apache/avro/tool/Main.java

```diff
@@ -104,18 +104,26 @@ private int run(String[] args) throws Exception {
 
   private static void printStream(InputStream in) throws Exception {
     byte[] buffer = new byte[1024];
-    for (int i = in.read(buffer); i != -1; i = in.read(buffer))
-      System.err.write(buffer, 0, i);
+    try {
+      for (int i = in.read(buffer); i != -1; i = in.read(buffer))
+        System.err.write(buffer, 0, i);
+    } finally {
+      in.close();
```

Comment:
This one is a private method and it seemed to be easier and shorter to close the stream inside this method instead of the public caller. Do you think I should rewrite it?